### PR TITLE
Added typescript definitions.

### PR DESCRIPTION
**Pull request by noah79**
I've added a .d.ts taken from the thread https://github.com/rackt/redux-simple-router/issues/151

I have not submitted this to definitelytyped.  However the definition I've included meets their header requirements.  Thanks for your contributions to this developer ecosystem.  It's bottled magic!

### CODE REVIEW

**Comment by Keats** [comment]
If you add a typings field in package.json pointing to the .d.ts, typescript should be able to pick it up automatically.
See https://github.com/Microsoft/TypeScript/wiki/Typings-for-npm-packages

**Comment by timdorr** [comment]
Also, can this be isolated to just code within the repo? The redux or react-router ones shouldn't need to be included. 

Also, is it possible (I'm not a TS user) to move this somewhere less conspicuous? I don't want to clutter up the tree with lots of extra directories for each feature or system. Can it just live at the top level in a definitions.ts file or something like that?

**Comment by rosendi** [comment]
@timdorr they shouldn't, i.e. the `redux-simple-router.d.ts` file under the `typings` directory or in the root is enough.

File: typings/redux-simple-router/redux-simple-router.d.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+///
+declare module ReduxSimpleRouter {
+    import R = Redux;
+    import H = HistoryModule;
+
+    const UPDATE_LOCATION: string;
+
+    interface HistoryMiddleware extends R.Middleware {
+        syncHistoryToStore(store: R.Store, selectRouterState?: Function): void;
```
Comment (Keats) [inline typings/redux-simple-router/redux-simple-router.d.ts]: it's called `listenForReplays` now afaik

**Comment by Keats** [comment]
And there's also https://github.com/DefinitelyTyped/DefinitelyTyped/pull/6629 with better typing for redux, which hasn't been merged yet

**Comment by timdorr** [comment]
@noah79 Any update on the necessary changes?

**Comment by noah79** [comment]
I've been busy with day job stuff.  I'll get back to this later this week.

**Comment by stepancar** [comment]
@Keats i think it's not good idea to support definitions in this repo, cause react-router-redux is written in js. Typescript npm packages is good if you write your package using ts. I think more better if enybody will push definitions to definetlyTyped repo

**Comment by rosendi** [comment]
We have to wait for this PR to get merged https://github.com/DefinitelyTyped/DefinitelyTyped/pull/6073

**Comment by Blystad** [comment]
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/6073 ( Defintitions for react-router v1.0.0 )  was marged in months ago. I guess you mean https://github.com/DefinitelyTyped/DefinitelyTyped/pull/6629 ( More specific type constraints [for Redux]) ?

**Comment by rosendi** [comment]
oops, I have mistaken. The default typings for React Router (react-router.d.ts) on DefinitelyTyped is pointed to 0.13.x, but should be 1.0.x. I think we can make a PR to the DefinitelyTyped repo now and do type constraint improvements later, can't we?

**Comment by rosendi** [comment]
I've just checked. My fork of DefinitelyTyped repo was outdated.

**Comment by timdorr** [comment]
Sooooo, should we keep this in DefinitelyTyped? I think so, and I'm going to close this, but feel free to make loud noises and I'll open it back up :smile:

**Comment by Ciantic** [comment]
Typings in DefinitelyTyped are not working with official typings of Redux.

It would be nice to have react-router-redux typings in the main repo and maintained, since [Redux has now official typings in the repo](https://github.com/reactjs/redux/blob/master/index.d.ts), right now I'm figuring out a way to make them work with official Redux typings.

**Pull request closed without merging**